### ISSUE TITLE
TabArea - We are experimenting with our AutoML pipeline called SmartML

### ISSUE THREAD

**Issue by DhavalRepo18** [TabArena/tabarena_benchmarking_examples#6]
Hi Author 

https://scholar.google.com/citations?view_op=view_citation&hl=en&user=c06_VJAAAAAJ&cstart=20&pagesize=80&sortby=pubdate&citation_for_view=c06_VJAAAAAJ:zA6iFVUQeVQC

We are creating this issue to let you know in advance we will touch base with you for additional help as well as a request to add the above work in related work section if possible for TabArena paper.

**Comment by LennartPurucker** [comment]
Hi,

Thank you for letting us know! 

As an AutoML system, your work is not directly related to our tabular *model* benchmark. 
I recommend taking a look and benchmarking your system with the AutoML benchmark: https://github.com/openml/automlbenchmark

Please feel free to elaborate on how far you see a connection, similar to the other papers we cite in our related work section.

I am otherwise closing this issue as it does not represent a problem that is to be fixed or a feature request.

**Comment by albertcthomas** [comment]
Sorry @LennartPurucker could you explain a bit more why you think this is not suited for AutoML system? Autogluon tabular seems to refer to this benchmark now and it is an AutoML system.

**Comment by LennartPurucker** [comment]
Heyho, thanks for letting me elaborate!

Ah, I think my first point might not have been clear enough. 

Let me reiterate and make it explicit: anyone is free to compare any tabular predictive modelling method to the leaderboard of TabArena. However, the current version of TabArena and its leaderboard aim to compare only model pipelines. 

In addition, the leaderboard includes reference pipelines to complement model comparisons. Our current definition of reference pipelines from the leaderboard (similar to the paper):

> The leaderboard includes a reference pipeline, which is applied independently of the tuning protocol and constraints we constructed for models within TabArena. The reference pipeline aims to represent the performance quickly achievable by a practitioner on a dataset. The current reference pipeline is the predictive machine learning system AutoGluon (version 1.3, with the best_quality preset and 4 hours for training). AutoGluon represents an ensemble pipeline across various model types and thus provides a reference for model-specific pipelines.

--- 

With this, let me reply to your comments one by one:

> Autogluon is relying on ensembles. I would have thought that if I have a black box model [...] this would be allowed? 

* To make this explicit as well: AutoGluon 1.3 is not a model in TabArena but a reference pipeline. 
* Sure, you are absolutely free to compare black box models to TabArena. The leaderboard could include such black box models as a reference pipeline in future iterations. In a distant future, we may also have a unique leaderboard for predictive (black box) machine learning systems. 

> TabArena would not know it is doing ensembles. [...] if everything is wrapped into a scikit-learn API, how would TabArena know that it is using more than one model type?

TabArena, in its current iteration, requires full open source mode for any results to be admitted to the leaderboard. Future leaderboards might change this. 

As we share all our result artifacts, anyone is free to compare their closed-source, black box models to the results on TabArena. Moreover, all our evaluation and plot code is public, such that anyone can reproduce the leaderboard and plots, including their new method, and have the same style and look as we use on the leaderboard or in the paper. The official, verified results will only be shared by us via the leaderboard at [tabarena.ai](https://tabarena.ai).


> but that here this is imposed by TabArena? 

Yes. Maybe this also became clearer from my comments above. 
For any model benchmark, I would assume we will always impose such restrictions as they are necessary design decisions. 

>  it should be made more clear that TabArena is not currently meant for AutoML

I was hoping that our paper and the leaderboard do this, e.g., see the second sentence on the leaderboard: "The goal of TabArena and its leaderboard is to assess the peak performance of model-specific pipelines.". 
Any thoughts on what could help to make this clearer or what you would like to see? 

---

Finally, to reiterate: anyone is free to compare models and reference pipelines to the results from TabArena. The results of our model pipelines from TabArena should be a (strong) baseline for any system. However, without comparing to other AutoML systems (or similar methods), any claim must be judged correctly. 

I would argue that in the case of the post you linked for AutoGluon 1.4, the comparison to 1.3 is sufficient, no? Moreover, a comparison to the SOTA models seems important, as many practitioners often rely on model-specific pipelines instead of ensemble pipelines. 

There is currently no public benchmark for AutoGluon 1.4 compared to other AutoML systems, such as FLAML, LightAutoML, or systems such as MLZero and Agent K. For the latter, TabArena-Agent is might be a cool name -- to seed some thoughts :) For such a comparison, I would recommend that anyone working on tabular IID data use TabArena's benchmark datasets, evaluation design, and model pipelines as baselines.
